aws_embedded_metrics/validator.py

```diff
@@ -114,3 +116,39 @@ def validate_namespace(namespace: str) -> None:
 
     if not re.match(constants.VALID_NAMESPACE_REGEX, namespace):
         raise InvalidNamespaceError(f"Namespace contains invalid characters: {namespace}")
+
+
+def validate_timestamp(timestamp: datetime) -> None:
+    """
+    Validates a given timestamp based on CloudWatch Timestamp guidelines.
+
+    Timestamp must meet CloudWatch requirements, otherwise a InvalidTimestampError will be thrown.
+    See [Timestamps](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#about_timestamp)
+    for valid values.
+
+    Parameters:
+        timestamp (datetime): Datetime object representing the timestamp to validate.
+
+    Raises:
+        InvalidTimestampError: If the timestamp is either None, too old, or too far in the future.
+    """
+    if not timestamp:
+        raise InvalidTimestampError("Timestamp must be a valid datetime object")
+
+    timestamp_past_age_error_message = f"Timestamp {str(timestamp)} must not be older than {str(constants.MAX_TIMESTAMP_PAST_AGE)} milliseconds"
+    timestamp_future_age_error_message = f"Timestamp {str(timestamp)} must not be newer than {str(constants.MAX_TIMESTAMP_FUTURE_AGE)} milliseconds"
+
+    if timestamp == datetime.min:
+        raise InvalidTimestampError(timestamp_past_age_error_message)
```

Comment:
Instead of having a check for `datetime.min` here, why don't we update the `convert_to_milliseconds` method to support this?

Reply:
Yes, this works too. Anything below datetime.min is always treated at 0 seconds in our case.